streams/runners/storm/actions/src/main/java/com/hortonworks/streamline/streams/actions/storm/topology/StormTopologyActionsImpl.java

```diff
@@ -490,6 +507,101 @@ private String createYamlFile (TopologyLayout topology) throws Exception {
+                "hbaseCredentialsConfigKeys", "org.apache.storm.hbase.security.AutoHBase");
+    }
+
+    private void putServiceSpecificCredentialConfig(Config topologyConfig, TopologyDag topologyDag,
+                                                    Map<String, Map<String, String>> clusterToConfiguration,
+                                                    List<Class<?>> outputComponentClasses,
+                                                    List<Class<?>> inputComponentClasses,
+                                                    String serviceName, String clusterKeyPrefix,
+                                                    String keytabPathKeyName, String principalKeyName,
+                                                    String credentialConfigKeyName,
+                                                    String topologyAutoCredentialClassName) {
+
+        boolean componentExists = false;
+        for (OutputComponent outputComponent : topologyDag.getOutputComponents()) {
+            for (Class<?> clazz : outputComponentClasses) {
+                if (outputComponent.getClass().isAssignableFrom(clazz)) {
+                    componentExists = true;
+                    break;
+                }
+            }
+        }
+
+        if (!componentExists) {
+            for (InputComponent inputComponent : topologyDag.getInputComponents()) {
+                for (Class<?> clazz : inputComponentClasses) {
+                    if (inputComponent.getClass().isAssignableFrom(clazz)) {
+                        componentExists = true;
+                        break;
+                    }
+                }
+            }
+        }
+
+        if (componentExists) {
+            List<String> clusterKeys = new ArrayList<>();
+
+            clusterToConfiguration.keySet()
+                    .forEach(clusterName -> {
+                        // FIXME: it shouldn't read service config if the service isn't belong to the namespace
+                        Map<String, String> conf = serviceConfigurationReader.read(clusterName, serviceName);
+                        // add only when such (cluster, service) pair is available for the namespace
+                        if (!conf.isEmpty()) {
+                            Map<String, String> confForToken = clusterToConfiguration.get(clusterName);
+                            conf.put(principalKeyName, confForToken.get("principal"));
+                            conf.put(keytabPathKeyName, confForToken.get("keytabPath"));
+
+                            String clusterKey = clusterKeyPrefix + clusterName;
+                            topologyConfig.put(clusterKey, conf);
+                            clusterKeys.add(clusterKey);
+                        }
+                    });
+
+            topologyConfig.put(credentialConfigKeyName, clusterKeys);
+
+            Optional<List<String>> autoCredentialsOptional = topologyConfig.getAnyOptional("topology.auto-credentials");
```

Comment:
Here we also add `topology.auto-credentials` when we add credential config key for the service.